### PR TITLE
chore(refactor): merge pages for address + contract + token + token contract

### PR DESCRIPTION
**Pull request by lykalabrada**
#### Which issue(s) does this PR fixes?:

Fixes DFC-133, DFC-135, DFC-125

#### Additional comments?:
Known issues:
- [x] Token holders next page params is giving big number value - causing pagination to fail
Sample token: `/token/0x0074b7eb472A1FB2c60915ddAf9B122ec5892e32`

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*

### CODE REVIEW

File: src/pages/address/_components/TokenTransfersList.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+}) {
+  const { connection } = useNetwork();
+  const [transfers, setTransfers] = useState<TokenTransferProps[]>([]);
+  const [nextPage, setNextPage] = useState<TokenTransferPageParamsProps>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [trigger] = useGetTokenTransfersMutation();
+  const router = useRouter();
+
+  const params = router.query;
+  const fetchTokenTransfers = async () => {
```
Comment (pierregee) [inline src/pages/address/_components/TokenTransfersList.tsx:33]: All of the status for token transfers list failed but successful in the detail page, hmm does the token transfers list able to provide the status of each transfer? 

image
image

File: src/pages/address/_components/TokenTransfersList.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        transfers.map((item) => (
+          <TransactionRow
+            key={item.tx_hash}
+            rawData={
+              {
+                ...item,
+                hash: item.tx_hash,
+                tx_types: [item.type],
+                token_transfers: [item],
+              } as any
```
Comment (pierregee) [inline src/pages/address/_components/TokenTransfersList.tsx]: For maintainability, can we add proper types?

File: src/pages/address/_components/contract/ReadWriteContract.tsx [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
-import { useRouter } from "next/router";
 import { ContractMethodType } from "@api/types";
 import { useNetwork } from "@contexts/NetworkContext";
 import { useGetContractMethodsQuery } from "@store/contract";
 import { FiLoader } from "react-icons/fi";
 import LinkText from "@components/commons/LinkText";
 import ContractMethod from "./ContractMethod";
-import ConnectButton from "../ConnectButton";
+import ConnectButton from "./ConnectButton";
+// import ConnectButton from "../ConnectButton";
```
Comment (pierregee) [inline src/pages/address/_components/contract/ReadWriteContract.tsx]:
```suggestion
```

File: src/pages/address/_components/contract/ReadWriteContract.tsx [hunk 4]
```diff
@@ -1,30 +1,32 @@
...
+  addressHash: string;
   implementationAddress: string | null;
 }) {
   const [expandAll, setExpandAll] = useState<boolean>(false);
   const [resetForm, setResetForm] = useState<boolean>(false);
   const { connection } = useNetwork();
-  const router = useRouter();
-  const cid = router.query.cid?.toString()!;
+  // const router = useRouter();
+  // const cid = router.query.cid?.toString()!;
```
Comment (pierregee) [inline src/pages/address/_components/contract/ReadWriteContract.tsx]:
```suggestion
```

File: src/pages/tx/[tid].tsx [hunk 5]
```diff
@@ -217,7 +217,7 @@ export default function Transaction({
                   label={truncateTextFromMiddle(txDetails.to, 4)}
                   href={
                     txDetails.isToContract
-                      ? `/contract/${txDetails.to}`
+                      ? `/address/${txDetails.to}`
```
Comment (pierregee) [inline src/pages/tx/[tid].tsx]: can remove the ternary, same value on both truthy and falsy

File: src/pages/tx/[tid].tsx [hunk 6]
```diff
@@ -192,7 +192,7 @@ export default function Transaction({
                 label={truncateTextFromMiddle(txDetails.from, 4)}
                 href={
                   txDetails.isFromContract
-                    ? `/contract/${txDetails.from}`
+                    ? `/address/${txDetails.from}`
```
Comment (pierregee) [inline src/pages/tx/[tid].tsx]: can remove the ternary, same value on both truthy and falsy

**Review by pierregee — CHANGES_REQUESTED** [review]

File: src/pages/address/_components/TokenTransfersList.tsx (hunk 1 above)
Comment (lykalabrada) [reply src/pages/address/_components/TokenTransfersList.tsx:33]: Just checked, so no we cannot get the status for each transfer using the same api. We might have to remove status for "Token transfers"

File: src/pages/address/_components/TokenTransfersList.tsx (hunk 2 above)
Comment (lykalabrada) [reply src/pages/address/_components/TokenTransfersList.tsx]: Updated

File: src/pages/address/_components/TokenTransfersList.tsx (hunk 1 above)
Comment (lykalabrada) [reply src/pages/address/_components/TokenTransfersList.tsx:33]: Removed the status for this list

File: src/components/TransactionDetails.tsx [hunk 7]
```diff
@@ -86,9 +84,24 @@ export default function TransactionDetails({
...
+                    hash: transformedData.hash,
+                    amount: transformedData.amount,
+                    symbol: transformedData.symbol,
+                    timeInSec: transformedData.timeInSec,
+                    status: transformedData.status,
+                  }}
+                />
+              );
+            })
           )}
```
Comment (fullstackninja864) [inline src/components/TransactionDetails.tsx]:
```suggestion
            transactions.map((item) => (
                <TransactionRow
                  key={item.hash}
                  data={transformTransactionData(item)}
                />
              )
            )
          )}
```

File: src/pages/address/[aid].tsx [hunk 8]
```diff
@@ -1,79 +1,460 @@
...
+  const detailContainerCss =
+    "flex justify-between md:justify-start md:flex-col gap-1";
+  const detailValueCss = "text-white-50 break-all text-right md:text-start";
 
   return (
     <div className="px-1 md:px-0 mt-12">
       <SearchBar containerClass="mt-1 mb-6" />
       <GradientCardContainer className="relative z-[1]" fullBorder>
-        <div className="lg:p-10 md:p-10 px-5 py-6">
+        <div className="lg:p-10 md:p-10 px-5 py-8">
```
Comment (fullstackninja864) [inline src/pages/address/[aid].tsx]:
```suggestion
        <div className="px-5 py-8 md:p-10">
```

File: src/pages/txs/index.tsx [hunk 9]
```diff
@@ -66,9 +64,24 @@ export default function Transactions({
...
+                    to: transformedData.to,
+                    hash: transformedData.hash,
+                    amount: transformedData.amount,
+                    symbol: transformedData.symbol,
+                    timeInSec: transformedData.timeInSec,
+                    status: transformedData.status,
+                  }}
+                />
+              );
+            })
```
Comment (fullstackninja864) [inline src/pages/txs/index.tsx]:
```suggestion
            data.transactions.map((tx) => (
                <TransactionRow
                  key={tx.hash}
                  data={transformTransactionData(tx)}
                />
              )
            )
```

**Review by fullstackninja864 — CHANGES_REQUESTED** [review]

File: src/components/TransactionDetails.tsx (hunk 7 above)
Comment (lykalabrada) [reply src/components/TransactionDetails.tsx]: type is not the same as what `transformTransactionData` returns 😅

File: src/pages/txs/index.tsx (hunk 9 above)
Comment (lykalabrada) [reply src/pages/txs/index.tsx]: same as above, `data`'s type is not the same as what `transformTransactionData` returns

Comment (lykalabrada) [reply src/pages/txs/index.tsx]: actually you're right, it works this way. Thank you! 🎉

File: src/pages/address/_components/TokenHoldersList.tsx [hunk 10]
```diff
@@ -40,35 +40,32 @@ export default function TokenHoldersList({
     }).unwrap();
     setHolders(data.items);
     setNextPage(data.next_page_params);
+    await sleep(150);
```
Comment (pierregee) [inline src/pages/address/_components/TokenHoldersList.tsx:43]: can we add a comment on why this is added (flicker issue, right?)

**Review by pierregee — APPROVED** [review]

File: src/pages/address/_components/TokenHoldersList.tsx (hunk 10 above)
Comment (pierregee) [reply src/pages/address/_components/TokenHoldersList.tsx:43]: can add in another pr

**Pull request merged**